app/scripts/modules/kubernetes/src/rawResource/controller/FiltersPubSub.ts

```diff
@@ -0,0 +1,35 @@
+import { IK8sResourcesFiltersState } from '../component/K8sResourcesFilters';
+
+type ISubscriber = (message: IK8sResourcesFiltersState) => void;
+
+export class FiltersPubSub {
+  private subscribers: ISubscriber[] = [];
+  private static intances: Record<string, FiltersPubSub> = {};
+
+  private constructor() {}
+
+  public static getInstance(name: string) {
+    let instance = FiltersPubSub.intances[name];
+    if (!FiltersPubSub.intances[name]) {
+      FiltersPubSub.intances[name] = new FiltersPubSub();
+      instance = FiltersPubSub.intances[name];
+    }
+    return instance;
+  }
+
+  public subscribe(subscriber: ISubscriber) {
+    this.subscribers.push(subscriber);
+    return () => (this.subscribers = this.subscribers.filter((x) => x !== subscriber));
+  }
+
+  public unsubscribe(subscriber: ISubscriber) {
+    const index = this.subscribers.indexOf(subscriber);
+    if (index > -1) {
+      this.subscribers.splice(index, 1);
+    }
+  }
+
+  public publish(message: IK8sResourcesFiltersState) {
```

Comment:
I'd recommend using a different interface to type `message` which contains only the data that is relevant to subscribers.  This avoids leaking internal component state and abstractions.

If subscribers need _everything_ in `IK8sResourcesFiltersState` then disregard; this is fine.